units/crons.go

```diff
@@ -248,6 +248,37 @@ func PopulateHostTerminationJobs(env evergreen.Environment) amboy.QueueOperation
 	}
 }
 
+func PopulateIdleHostJobs(env evergreen.Environment) amboy.QueueOperation {
+	return func(queue amboy.Queue) error {
+		flags, err := evergreen.GetServiceFlags()
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		if flags.MonitorDisabled {
+			grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
+				"message": "monitor is disabled",
+				"impact":  "not submitting detecting idle hosts",
+				"mode":    "degraded",
+			})
+			return nil
+		}
+
+		catcher := grip.NewBasicCatcher()
+		ts := util.RoundPartOfHour(1).Format(tsFormat)
+		hosts, err := host.AllIdleEphemeral()
+		if err != nil {
+			return errors.Wrap(err, "problem finding idle hosts")
+		}
+
+		for _, h := range hosts {
+			catcher.Add(queue.Put(NewIdleHostTerminationJob(env, h, ts)))
```

Comment:
Would it be better to move the idle time checking here, and only add a host termination job if we want to terminate it? That way we won't add a bunch of no-op jobs to the queue

Reply:
for distros with hourly billing, we have to do aws queries and interact with the host provider. We could cache more of this information in the database and then add a clause to the `host.FindHostsToTerminate()` query, but for now I think we need to do some of this this logic application side.

noop jobs aren't really a problem?